### DISCUSSION TITLE
express run server.ts

### DISCUSSION THREAD

**Discussion by EvgeniyBudaev** [remix-run/remix#9054] [Q&A]
I ask you to help me start the server.ts on express. I tried many options and nothing works.
https://github.com/EvgeniyBudaev/remixjs-2-8

npm run dev 

![2024-03-14_18-32-57](https://github.com/remix-run/remix/assets/49036840/9d5b1a89-57ab-4a35-a034-b7af7d029bff)

**Comment by lili21** [comment]
how you tried just using javascript instead of typescript for server.js?

**Reply by lili21** [reply]
You can check this https://github.com/remix-run/remix/discussions/6004 out. `remix dev` doesn't support ts yet.

**Comment by kiliman** [comment]
I've created an example. 

https://github.com/kiliman/remix-vite-express

The trick is to compile the server.ts file using esbuild. 

https://github.com/kiliman/remix-vite-express/blob/6451bdd0e9062e49956ab59b578465546170c7f1/package.json#L6:L12

**Reply by EvgeniyBudaev** [reply]
How to import files? I get the Error [ERR_MODULE_NOT_FOUND]
![2024-03-17_08-17-27](https://github.com/remix-run/remix/assets/49036840/58eebcbc-5a35-48c3-a66b-dc41d23c67be)
![2024-03-17_08-17-18](https://github.com/remix-run/remix/assets/49036840/e6355336-6d78-41d0-b598-2af9d4c1a454)
![2024-03-17_08-16-57](https://github.com/remix-run/remix/assets/49036840/9760fd7e-8867-4e44-b7c6-d14fe58c15db)
